sqlglot/dialects/presto.py

```diff
@@ -186,6 +186,12 @@ def _unix_to_time_sql(self: Presto.Generator, expression: exp.UnixToTime) -> str
     return ""
 
 
+def _to_int(expression: exp.Expression) -> exp.Expression:
+    if expression.type and expression.type.this not in exp.DataType.INTEGER_TYPES:
+        return exp.cast(expression.copy(), to=exp.DataType.Type.BIGINT)
```

Comment:
do you need the cast here?

Reply:
you shouldn't copy, here, it's a double copy

Reply:
This is the whole point of the change, right?

Reply:
ah, copy, not cast. whoopsie